### PR TITLE
Adds time-chunked adjoint passes

### PR DESCRIPTION
**Pull request by reverendbedford**
Extends the time chunking to the adjoint/backward pass.

### CODE REVIEW

File: pyoptmat/ode.py [hunk 1]
```diff
@@ -104,27 +107,28 @@ def update_adjoint(self, dt, J, a_prev, grads):
...
-        # This has to be done sequentially...
-        for i in range(ntime):
-            adjoint_block[i + 1] = torch.linalg.lu_solve(
-                lu[i], pivot[i], adjoint_block[i].unsqueeze(-1)
-            ).squeeze(-1)
-            adjoint_block[i + 1] += grads[i + 1]
-
-        return adjoint_block
+        # Actually do the batch solve
+        adjoint_block[1:] = op.matvec(rhs).squeeze(-1)
```
Comment (hugary1995) [inline pyoptmat/ode.py:130]: why do we need a squeeze here?

File: pyoptmat/chunktime.py [hunk 2]
```diff
@@ -387,7 +377,7 @@ def forward(self, v):
                 v[i] - torch.bmm(self.B[i - 1], v[i - 1].clone()),
```
Comment (hugary1995) [inline pyoptmat/chunktime.py:377]: why clone()?

Comment (reverendbedford) [reply pyoptmat/chunktime.py:377]: It should not be necessary but `torch.autograd` throws an "in place" operation complaint without it.  I'll add a comment.

File: pyoptmat/ode.py (hunk 1 above)
Comment (hugary1995) [reply pyoptmat/ode.py:130]: > Also: it doesn't work when I take it off...

That's it!

**Review by hugary1995 — APPROVED** [review]

**Pull request merged**